awc/src/connect.rs

```diff
@@ -103,17 +107,41 @@ where
 
     fn call(&self, req: ConnectRequest) -> Self::Future {
         // connect to the host
-        let fut = match req {
-            ConnectRequest::Client(ref head, .., addr) => self.connector.call(ClientConnect {
-                uri: head.as_ref().uri.clone(),
-                addr,
-            }),
-            ConnectRequest::Tunnel(ref head, addr) => self.connector.call(ClientConnect {
-                uri: head.uri.clone(),
-                addr,
-            }),
+        let (head, addr, sni_host) = match req {
+            ConnectRequest::Client(ref head, .., addr, ref sni_host) => {
+                (head.as_ref(), addr, sni_host.clone())
+            }
+            ConnectRequest::Tunnel(ref head, addr, ref sni_host) => (head, addr, sni_host.clone()),
+        };
+
+        let authority = if let Some(authority) = head.uri.authority() {
+            authority
+        } else {
+            return ConnectRequestFuture::Error {
```

Comment:
there may be a better way, but a lot of code need to be updated and was not sure about the correct / standard thing to do here, this is a pragmatic way of doing it, but if you have an idea on how to avoid this will be glad to hear.